crates/synthesis/src/multi_controlled/mcx.rs

```diff
@@ -238,13 +237,13 @@ impl CircuitDataForSynthesis for CircuitData {
 #[pyfunction]
 pub fn c4x() -> PyResult<CircuitData> {
     let mut circuit = CircuitData::with_capacity(5, 0, 0, Param::Float(0.0))?;
-    circuit.h(4);
-    circuit.cp(PI2, 3, 4);
-    circuit.h(4);
+    circuit.h(4)?;
```

Comment:
It feels weird to me that this kind of thing can fail. Though I suppose that's partly why we keep this experimental circuit builder stuff private to `mcx.rs` for now.

Long term, I would expect that any instruction-specific append methods on a `CircuitData` would not raise typing-related errors since their signatures would be specific to the instruction in question.